portal/static/js/main.js

```diff
@@ -2592,7 +2592,53 @@ var Profile = function(subjectId, currentUserId) {
     };
     this.initPatientEmailFormSection = function() {
         var self = this;
-        if (!hasValue($("#email").val())) $("#profileEmailSelect").attr("disabled", true);
+        if (!$("#email").val()) {
+            $("#profileEmailSelect").attr("disabled", true);
+        };
+
+        if ($("#assessmentStatusContainer").length > 0) {
+            tnthAjax.patientReport(self.subjectId, function(data) {
+                if (!data.error) {
+                    var pcpReports = data["user_documents"]? $.grep(data["user_documents"], function(document) {
+                        return /P3P/gi.test(document.filename);
+                    }): false;
+                    var hasReports = pcpReports && pcpReports.length > 0;
+                    /*
+                     * adjust select/send email UI only if email is available
+                     */
+                    if (hasValue($("#email").val())) {
+                        if (!hasReports) {
+                            $("#lbPatientRegEmail").addClass("active");
+                        } 
+                    }
+                    /*
+                     * show reports link to user if completed assessment
+                     */
+                    if (hasReports) {
+                        $(".email-selector-container").html("<a href='#profilePatientReportTable' class='report-link'>" + i18next.t("View reports") + "</a>");
+                        $("#btnProfileSendEmail").hide();
+                    }
+                    $("#assessmentStatusContainer .assessment-label")
+                    .html(hasReports?i18next.t("complete"):i18next.t("incomplete"))
+                    .addClass(hasReports?"text-success":"text-warning")
+                    .removeClass(!hasReports?"text-success": "text-warning");
+                }
+            });
+        }
+
+        if ($("#registrationStatusContainer").length > 0) {
+            tnthAjax.getRoles(self.subjectId, false, function(data) {
```

Comment:
It might be convenient to implement a function (or extend `getRoles()`) so you can pass a specific role and get a true/false return instead of having to filter the results every time.